### ISSUE TITLE
PS 1.7 email alerts when product available not working

### ISSUE THREAD

**Issue by omar2886** [PrestaShop/PrestaShop#19611]
PS 1.7.6
PS-EMAILALERTS v2.1.1

Hi,
I have an issue with this module, it's not sending emails to customers when product becomes available.
I've checked table mailalert_customer_oos and it's well registering the emails, also checked /var/logs/prod.log and there are no errors. So I don't know what's happening, this module also deals with other email alerts to admin that are working fine.
I also have to say that I modified the code for the module to correctly understand when to consider a product as available or not, because in general it's misleading the way PrestaShop does it by default (we've talked about it on this topic [#14518](https://github.com/PrestaShop/PrestaShop/issues/14518)). Just so you know what I did thanks to Ventura:

**File:** /modules/ps_emailalerts/ps_emailalerts.php
**Hook modified:** hookDisplayProductAdditional
`if (!$this->customer_qty || !Configuration::get('PS_STOCK_MANAGEMENT') || $stockAvailable > 0) return;`
**Other Hooks:** hookActionUpdateQuantity and hookActionProductAttributeUpdate
```
if ($this->customer_qty && $result['quantity'] > 0) {

MailAlert::sendCustomerAlert((int) $result['id_product'], (int) $params['id_product_attribute']);

}
```
Now the module shows well on the product page when AvailableStock <=0. But the email alerts are not being sent, hope you can help!

**Comment by NeOMakinG** [comment]
Hello @omar2886,

It looks like I've been able to reproduce it and I don't receive any email but I've the alerts on my customer dashboard of this module.

Thanks

**Comment by marionf** [comment]
@NeOMakinG it's a regression of which version ?

**Comment by NeOMakinG** [comment]
Reproduced on 1.7.5 and 1.7.6.5, should not be considered as a regression I think, my bad

**Comment by omar2886** [comment]
Seems it's been released a new version for this module, I just installed yesterday but still not  working.
Also I've noticed that part of the codes has changed, so when I try to modify the 3 hooks I mentioned at the opening of this post, it doesn't work as expected. The desired behavior is that the block for a customer tu add his email to be notified when product is back in stock, only shows when Product Quantity =< 0. But now with Email Alers v2.2.0, when doing this code modification on the 3 hooks, the block will always appear no matter the stock level.

**Comment by hibatallahAouadni** [comment]
Hello @omar2886 

While reproducing this issue #22628 I notice that your problem is solved, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/103406001-6692c780-4b59-11eb-8d3a-14fe139dbf83.png)

So, I close this issue cause it's fixed with ps_emailalerts v2.2.1

PS: Please upgrade your module and check. If you encounter any issue, please notify us :wink: and remember that even the issue is closed, we can always answer all your questions about this issue :slightly_smiling_face:  

Thanks!
